nvflare/private/fed/app/simulator/simulator_runner.py

```diff
@@ -532,8 +532,12 @@ def run_client_thread(self, num_of_threads, gpu, lock, timeout=60):
                 client.simulate_running = False
         except Exception as e:
             self.logger.error(f"run_client_thread error: {secure_format_exception(e)}")
+        finally:
+            if rank == 0:
```

Comment:
The "self.federated_clients" is shared across all threads (the name called threads, but it is actually "process")

So we just need to iterate through them once, rank ==0 or rank == 1 either would work.
Just a condition to avoid END_RUN event to be fired multiple times.